wordpress-mobile/WordPress-iOS
# Fixing Today widget configuration on account/site changes

Fixes #2598, #2746, and #4687 

To test:

1. Log out of your WordPress.com account and verify that the widget is not configured to a site anymore.
2. Delete the site that the Today widget is configured to. If there are no sites, the widget is unconfigured.
3. Delete the site that the Today widget is configured to. If there is another site, the widget is configured to one of the other sites.
4. Open the widget when there is no internet connection and verify that it stays configured.
5. Open the widget when there is an invalid oauth token (you can set it manually in `TodayViewController.swift` line 165)

Change 
`let statsService: WPStatsService = WPStatsService(siteId: siteID, siteTimeZone: timeZone, oauth2Token: oauthToken, andCacheExpirationInterval:0)` 

to 

`let statsService: WPStatsService = WPStatsService(siteId: siteID, siteTimeZone: timeZone, oauth2Token: "fds", andCacheExpirationInterval:0)`

and verify that the today widget becomes unconfigured.

Needs review: @astralbodies
